applications/TrilinosApplication/tests/test_trilinos_linear_solvers.py

```diff
@@ -517,7 +689,14 @@ def test_amgcl_mpi_solver_gmres(self):
                     }
                 ]
             }
-            """)
+            """
+
+        with self.subTest('All ranks (MPI_COMM_WORLD)'):
+            self._RunParametrized(params_string)
+
+        with self.subTest('SubComm'):
+            self.skipTest("AMGCL does not yet support SubCommunicators")

```

Comment:
note that AMGCL internally hardcodes to use `MPI_COMM_WORLD`, hence this cannot be enabled yet

we need to refactor this